cmd/internal/apidump/apidump.go

```diff
@@ -87,22 +93,51 @@ var Cmd = &cobra.Command{
 			}
 		}
 
+		// Allow specification of an alternate rotation time, default 1h.
+		// But, if the trace name is explicitly given (or selected by tag) then
+		// we cannot rotate.
+		var traceRotateInterval time.Duration
+		if traceRotateFlag != "" {
+			if outFlag.AkitaURI.ObjectName != "" {
+				return errors.New("Cannot specify trace rotation along with a specific trace.")
+			}
+			traceRotateInterval, err = time.ParseDuration(traceRotateFlag)
+			if err != nil {
+				return errors.Wrap(err, "Failed to parse trace rotation interval.")
+			}
+		} else {
+			if outFlag.AkitaURI.ObjectName == "" {
+				traceRotateInterval = time.Hour
+			} else {
+				traceRotateInterval = time.Duration(0)
+			}
+		}
+
+		if deploymentFlag != "default" && os.Getenv("AKITA_DEPLOYMENT") != deploymentFlag {
+			printer.Stderr.Warningf("Deployment in environment variable %q overrides the command line value %q.",
+				deploymentFlag,
+				os.Getenv("AKITA_DEPLOYMENT"),
+			)
```

Comment:
I think you're right but I'm going to leave this as-is, because we introduced the environment variable first.

(However, I admit I'm partially motivated by not wanting to make a bigger change and hoist the handling of the environment variable up to this level from where it is currently set.)